package.json

```diff
@@ -91,19 +90,24 @@
     "@babel/core": "^7.19.0",
     "@babel/eslint-parser": "^7.18.9",
     "@babel/plugin-proposal-class-properties": "^7.18.6",
+    "@eslint/js": "^9.4.0",
     "@jest/globals": "^29.0.2",
+    "@types/eslint__js": "^8.42.3",
     "@types/react": "^18.2.73",
     "babel-plugin-lodash": "^3.2.11",
     "benchmark": "^2.1.4",
     "eslint": "^8.23.0",
-    "eslint-plugin-jest": "^27.0.1",
+    "eslint-config-next": "^14.1.4",
```

Comment:
What are dev-dependencies in the context of `nextstrain.org`? Building the site without these dev dependencies installed will cause next.js to download a bunch of them itself (potentially with different pins), which just seems worse, and then fail to build because of missing dependencies, which is categorically worse.

Maybe some could be classed as developer-only (e.g. `nodemon`) but anything that's necessary for CI to pass or the site to run seems like it should be a proper dependency.

Reply:
The [node package.json docs](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#devdependencies) have this to say about `devDependencies`:

> If someone is planning on downloading and using your module in their program, then they probably don't want or need to download and build the external test or documentation framework that you use.

The [dependencies section](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#devdependencies) also says: 

> Please do not put test harnesses or transpilers in your dependencies object. See devDependencies, below.

Linting config feels (to me) very akin to "test harnesses or transpilers" — it's only something that's going to be needed if you're working on the code. 

As far as CI, etc — `npm ci` installs devDeps (as long as the lock file that's present was built with a normal `npm install` and not `npm install --production). Additionally the CI GitHub workflow does some extra dance steps to make sure devDeps are installed. 

I don't care about this _super_ strongly, but it doesn't make a whole lot of sense to have several linting related packages (`eslint`, `eslint-plugin-jest`, `typescript-eslint`, etc) in `devDeps` and one (`eslint-config-next`) as a regular dep, does it?

Reply:
If it were up to me I'd put everything in dependencies and call it a day. But I don't want us to get sidetracked on the minutiae so... let's consider this resolved.

Reply:
I would probably be inclined to +1 a PR that did that :grin: …but I wanted to land the linting changes, so I went ahead and merged.